app/routes/dashboard.apps.$appId.notifications/components/NotificationsAlertForm/NotificationsAlertForm.tsx

```diff
@@ -65,67 +28,83 @@ export default function NotificationsAlertForm({
 }: NotificationsAlertFormProps) {
   const navigation = useNavigation()
   const { notificationSettings } = endpoint
+  const [changeHappened, setChangeHappened] = useState(false)
```

Comment:
I think we can use a better name here. Maybe something that aligns with form usage and terminology, like  `formDirty` / `formUpdated`/ `formTouched` /`isEditing` ...etc 

Or maybe we can use Foorm validators that come out of the box from Remix itself. and use `touchedFields`https://www.remix-validated-form.io/reference/use-form-context#touchedFields

Reply:
I think that's a separate library that doesn't come with remix